timed/reports/views.py

```diff
@@ -268,11 +268,16 @@ def _create_workreport(self, from_date, to_date, today, project, reports, user):
 
     def list(self, request, *args, **kwargs):
         queryset = self.filter_queryset(self.get_queryset())
+
+        if queryset.count() == 0:
+            return HttpResponseBadRequest()
+
         # needed as we add items in reverse order to work report
         queryset = queryset.reverse()
-        count = queryset.count()
-        if count == 0:
-            return HttpResponseBadRequest()
+
+        if settings.WORK_REPORTS_EXPORT_MAX_COUNT > 0:
+            queryset = queryset[: settings.WORK_REPORTS_EXPORT_MAX_COUNT]
```

Comment:
From a business perspective, this could be problematic. You're silently removing a part from the output without the user noticing, which could lead to problems down the road, depending on what's being done with the report.

Instead, I'd prefer to have a 400 bad request with the message to filter the data some more to get below the max row count.

(The message of course would need to be shown in the frontend as well)

Reply:
Valid point, that came to my mind too

Reply:
done